SpatialGDK/Source/SpatialGDK/Private/Utils/EntityFactory.cpp

```diff
@@ -0,0 +1,351 @@
+	// Actors with bNetLoadOnClient=false also need a StablyNamedObjectRef for linking in the case of loading from a snapshot or the server crashes and restarts.
+	TSchemaOption<FUnrealObjectRef> StablyNamedObjectRef;
+	TSchemaOption<bool> bNetStartup;
+	if (Actor->HasAnyFlags(RF_WasLoaded) || Actor->bNetStartup)
+	{
+		// Since we've already received the EntityId for this Actor. It is guaranteed to be resolved
+		// with the package map by this point
+		FUnrealObjectRef OuterObjectRef = PackageMap->GetUnrealObjectRefFromObject(Actor->GetOuter());
+		if (OuterObjectRef == FUnrealObjectRef::UNRESOLVED_OBJECT_REF)
+		{
+			FNetworkGUID NetGUID = PackageMap->ResolveStablyNamedObject(Actor->GetOuter());
+			OuterObjectRef = PackageMap->GetUnrealObjectRefFromNetGUID(NetGUID);
+		}
+
+		// No path in SpatialOS should contain a PIE prefix.
+		FString TempPath = Actor->GetFName().ToString();
+		GEngine->NetworkRemapPath(NetDriver, TempPath, false /*bIsReading*/);
+
+		StablyNamedObjectRef = FUnrealObjectRef(0, 0, TempPath, OuterObjectRef, true);
+		bNetStartup = Actor->bNetStartup;
+	}
+
+	TArray<Worker_ComponentData> ComponentDatas;
+	ComponentDatas.Add(Position(Coordinates::FromFVector(GetActorSpatialPosition(Actor))).CreatePositionData());
+	ComponentDatas.Add(Metadata(Class->GetName()).CreateMetadataData());
+	ComponentDatas.Add(SpawnData(Actor).CreateSpawnDataData());
+	ComponentDatas.Add(UnrealMetadata(StablyNamedObjectRef, ClientWorkerAttribute, Class->GetPathName(), bNetStartup).CreateUnrealMetadataData());
+
+	if (!Class->HasAnySpatialClassFlags(SPATIALCLASS_NotPersistent))
+	{
+		ComponentDatas.Add(Persistence().CreatePersistenceData());
+	}
+
+	if (SpatialSettings->bEnableUnrealLoadBalancer)
+	{
+		ComponentDatas.Add(AuthorityIntent::CreateAuthorityIntentData(NetDriver->VirtualWorkerTranslator->GetLocalVirtualWorkerId()));
+	}
+
+	if (Class->HasAnySpatialClassFlags(SPATIALCLASS_Singleton))
+	{
+		ComponentDatas.Add(Singleton().CreateSingletonData());
+	}
+
+	if (Actor->bAlwaysRelevant)
+	{
+		ComponentDatas.Add(AlwaysRelevant().CreateData());
+	}
+
+	if (Actor->NetDormancy >= DORM_DormantAll)
+	{
+		ComponentDatas.Add(Dormant().CreateData());
+	}
+
+	if (Actor->IsA<APlayerController>())
+	{
+#if !UE_BUILD_SHIPPING
+		ComponentDatas.Add(ComponentFactory::CreateEmptyComponentData(SpatialConstants::DEBUG_METRICS_COMPONENT_ID));
+#endif // !UE_BUILD_SHIPPING
+		ComponentDatas.Add(Heartbeat().CreateHeartbeatData());
+	}
+
+	ComponentFactory DataFactory(false, NetDriver);
+
+	FRepChangeState InitialRepChanges = Channel->CreateInitialRepChangeState(Actor);
+	FHandoverChangeState InitialHandoverChanges = Channel->CreateInitialHandoverChangeState(Info);
+
+	TArray<Worker_ComponentData> DynamicComponentDatas = DataFactory.CreateComponentDatas(Actor, Info, InitialRepChanges, InitialHandoverChanges);
+	ComponentDatas.Append(DynamicComponentDatas);
+
+	InterestFactory InterestDataFactory(Actor, Info, NetDriver->ClassInfoManager, NetDriver->PackageMap);
```

Comment:
```suggestion
	InterestFactory InterestDataFactory(Actor, Info, ClassInfoManager, PackageMap);
```

Reply:
This is a minor thing for now though. I understand your intention was to move the code from SpatialSender without any modifications.

Reply:
Addressed in 45c78cb; since I was making the change to pass the `ClassInfoManager` and `PackageMapClient` directly to the factory anyway, it made sense to fix this up as well :)